### PR TITLE
Using latest .Net SDK with small modifications

### PR DESCRIPTION
**Pull request by gucalder**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThrough` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/ElasticPoolCrudTests.ps1 [hunk 1]
```diff
@@ -171,9 +171,10 @@ function Test-GetElasticPoolMetric
     Assert-NotNull $ep1
     
     # Get pool metrics with all values
-    $metrics = $ep1 | Get-AzureRmMetric -TimeGrain "0:5:0" -StartTime "2015-04-22T16:00:00Z" -EndTime "2015-04-22T17:00:00Z"
-    Assert-NotNull $metrics
-    Assert-True { $metrics.Count -gt 0 }
+    # Commenting this section of the test becuase the GetAzureRmMetric has been replaced and must be called using -MetricNames <list of names> as parameter to enabled all the other parameters
```
Comment (gucalder) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/ElasticPoolCrudTests.ps1]: Oh yes. If I keep the old API the test should not fail. I forgot to revert this.
I'll revert this.

File: src/ResourceManager/Insights/Commands.Insights/Autoscale/NewAzureRmAutoscaleRuleCommand.cs [hunk 2]
```diff
@@ -29,15 +29,6 @@ public class NewAzureRmAutoscaleRuleCommand : AzureRMCmdlet
         private readonly TimeSpan MinimumTimeWindow = TimeSpan.FromMinutes(5);
         private readonly TimeSpan MinimumTimeGrain = TimeSpan.FromMinutes(1);
 
-        /// <summary>
-        /// Scale type enum.
-        /// Keep this enum for backwards compatibility. The only value supported is ChangeCount
-        /// </summary>
-        public enum ScaleType
```
Comment (gucalder) [inline src/ResourceManager/Insights/Commands.Insights/Autoscale/NewAzureRmAutoscaleRuleCommand.cs]: The value now comes from the SDK, it supports these values: ChangeCount (default), PercentChangeCount, ExactCount.
Line 99 of the file documents this, and line 117 warns the user that there are two more accepted values now.

File: documentation/breaking-changes/upcoming-breaking-changes.md [hunk 3]
```diff
@@ -25,4 +25,47 @@
     https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes/breaking-change-template.md
 -->
 
-# Upcoming Breaking Changes
\ No newline at end of file
+# Upcoming Breaking Changes
+
+## Release X.0.0 - January 2017
```
Comment (gucalder) [inline documentation/breaking-changes/upcoming-breaking-changes.md]: Ok, I just followed the template as it was.  
Date removed.

File: documentation/breaking-changes/upcoming-breaking-changes.md [hunk 4]
```diff
@@ -25,4 +25,103 @@
...
+    write-host "End time: $s11.EndTime"
+    write-host "Start time: $s11.StartTime"
+    write-host "ResourceId: $s11.ResourceId"
+    write-host "TimeGrain: $s11.TimeGrain"
+    write-host "Unit: $s11.Unit"
+    write-host "Metric values: $s11.MetricValues"
+
+    # New
+  
+    # This calls are valid. ResourceId remains mandatory and positional
```
Comment (gucalder) [inline documentation/breaking-changes/upcoming-breaking-changes.md]: I made some changes to this file. My goal here is to describe all the changes including the outputs, even if the calls remains the same as is the case with the Metric Definitions

Comment (gucalder) [inline documentation/breaking-changes/upcoming-breaking-changes.md]: I made some changes to this file. My goal here is to describe all the changes including the outputs, even if the calls remains the same as is the case with the Metric Definitions.
I also moved these breaking changes to the upcoming-breaking-changes.md in Insights\documentation

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/job/powershell-demand/1337/

**Pull request merged**